hazelcast/sql.py

```diff
@@ -0,0 +1,1629 @@
+import logging
+import uuid
+from threading import RLock
+
+from hazelcast.errors import HazelcastError
+from hazelcast.future import Future, ImmediateFuture
+from hazelcast.invocation import Invocation
+from hazelcast.util import (
+    UUIDUtil,
+    to_millis,
+    check_true,
+    get_attr_name,
+    try_to_get_error_message,
+    check_is_number,
+    check_is_int,
+    try_to_get_enum_value,
+)
+from hazelcast import six
+
+_logger = logging.getLogger(__name__)
+
+
+class SqlService(object):
+    """A service to execute SQL statements.
+
+    The service allows you to query data stored in a
+    :class:`Map <hazelcast.proxy.map.Map>`.
+
+    Warnings:
+
+        The service is in beta state. Behavior and API might change
+        in future releases.
+
+    **Querying an IMap**
+
+    Every Map instance is exposed as a table with the same name in the
+    ``partitioned`` schema. The ``partitioned`` schema is included into
+    a default search path, therefore a Map could be referenced in an
+    SQL statement with or without the schema name.
+
+    **Column resolution**
+
+    Every table backed by a Map has a set of columns that are resolved
+    automatically. Column resolution uses Map entries located on the
+    member that initiates the query. The engine extracts columns from a
+    key and a value and then merges them into a single column set.
+    In case the key and the value have columns with the same name, the
+    key takes precedence.
+
+    Columns are extracted from objects as follows (which happens on the
+    server-side):
+
+    - For non-Portable objects, public getters and fields are used to
+      populate the column list. For getters, the first letter is converted
+      to lower case. A getter takes precedence over a field in case of naming
+      conflict.
+    - For :class:`Portable <hazelcast.serialization.api.Portable>` objects,
+      field names used in the
+      :func:`write_portable() <hazelcast.serialization.api.Portable.write_portable>`
+      method are used to populate the column list.
+
+    The whole key and value objects could be accessed through special fields
+    ``__key`` and ``this``, respectively. If key (or value) object has fields,
+    then the whole key (or value) field is exposed as a normal field. Otherwise the
+    field is hidden. Hidden fields can be accessed directly, but are not returned
```

Comment:
I think the correct sentence is(as I told you in a message I sent to you): 

If key (or value) object has fields, then the key (or value) field is hidden. Otherwise, the whole key (or value) field is exposed as a normal field.